src/config.cpp

```diff
@@ -59,31 +59,126 @@ void parse_uint32(void* value_ptr, const char* txt) {
+        return config_option_type::uint32;
+    } catch (const std::invalid_argument&) {
+        // Fall through to string if integer conversion fails
+    }
+
+    // 3. If all else fails, assume it's a string
+    return config_option_type::string;
+}
+
+void read_config_file(std::unordered_map<std::string, std::string>& umap) {
+    std::string line;
+    std::ifstream config_stream;
+    // Get current working directory
+    auto current_path = std::filesystem::current_path();
+    // Create the full path to the config file
+    std::string full_config_path = (current_path / config_file).string();
+    config_stream.open(full_config_path);
+    if (!config_stream.is_open()) {
+        std::cerr << "Error opening config.toml" << std::endl;
+        return;
+    }
+
+    while (std::getline(config_stream, line)) {
+        // Ignore comments and empty lines
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+        line = trim(line);
+        // Check for section headers
+        if (line[0] == '[' && line[line.size() - 1] == ']') {
+            continue;
+        } else {
+            // Parse key-value pairs
+            size_t pos = line.find('=');
+            if (pos != std::string::npos) {
+                std::string key = trim(line.substr(0, pos));
+                std::string value = trim(line.substr(pos + 1));
+
+                // Store values (if any)
+                if (value != "") {
+                    umap[key] = value;
+                }
+            }
+        }
+    }
+    config_stream.close();
+}
+
 void parse_env() {
+    std::unordered_map<std::string, std::string> fileConfigValues;
+    read_config_file(fileConfigValues);
     for (auto& opt : gConfigOptions) {
         std::string var_name = "CLVK_";
         std::string optname_upper(opt.name);
         std::transform(optname_upper.begin(), optname_upper.end(),
                        optname_upper.begin(), ::toupper);
         var_name += optname_upper;
-        // printf("var_name = '%s' ", var_name.c_str());
         const char* txt = getenv(var_name.c_str());
         if (txt == nullptr) {
-            //    printf("is not set\n");
-            continue;
-        }
-        // printf("is set\n");
-        void* optval = const_cast<void*>(opt.value);
-        switch (opt.type) {
-        case config_option_type::string:
-            parse_string(optval, txt);
-            break;
-        case config_option_type::boolean:
-            parse_boolean(optval, txt);
-            break;
-        case config_option_type::uint32:
-            parse_uint32(optval, txt);
-            break;
+            if (fileConfigValues.find(opt.name) == fileConfigValues.end() ||
+                fileConfigValues[opt.name].length() == 0) {
+                continue;
+            }
+            auto curr_val = fileConfigValues[opt.name];
+            config_option_type optType = infer_type(fileConfigValues[opt.name]);
```

Comment:
Not sure I understand why you need that `infer_type` function at all. What's wrong with using `opt.type`?

Reply:
removed, thanks!